### PR TITLE
Add clr.aot subset for convenience and document it

### PR DESCRIPTION
**Pull request by agocke**

### CODE REVIEW

File: docs/workflow/building/coreclr/nativeaot.md [hunk 1]
```diff
@@ -2,6 +2,21 @@
 
 The Native AOT toolchain can be currently built for Linux (x64/arm64), macOS (x64) and Windows (x64/arm64).
 
+## Building
+
+1. [Install pre-requisites](../../README.md#build-requirements)
+1. Run `build[.cmd|.sh] clr.nativeaot+libs -rc [Debug|Release]` from the repo root to build binaries for local development. This will build individual components, but not the NuGet packages and builds much faster.
+
+The paths to major components can be overriden using `IlcToolsPath`, `IlcSdkPath`, `IlcFrameworkPath`, `IlcFrameworkNativePath` and `IlcMibcPath` properties for `dotnet publish`. For example, `/p:IlcToolsPath=<repo root>\artifacts\bin\coreclr\windows.x64.Debug\ilc` can be used to override the compiler with a local debug build for troubleshooting or quick iterations.
```
Comment (MichalStrehovsky) [inline docs/workflow/building/coreclr/nativeaot.md:10]: Would this paragraph make more sense in the "Building packages" section? It isn't immediately obvious these are instructions to consume the built bits.

File: eng/Subsets.props [hunk 2]
```diff
@@ -63,6 +63,8 @@
     <!-- Even on platforms that do not support the CoreCLR runtime, we still want to build ilasm/ildasm. -->
     <DefaultCoreClrSubsets Condition="'$(PrimaryRuntimeFlavor)' != 'CoreCLR'">clr.iltools+clr.packages</DefaultCoreClrSubsets>
 
+    <DefaultNativeAotSubsets>clr.alljits+clr.tools+clr.nativeaotlibs+clr.nativeaotruntime</DefaultNativeAotSubsets>
```
Comment (MichalStrehovsky) [inline eng/Subsets.props:66]: I think `UseNativeAotCoreLib` above might need a tweak. It controls what CoreLib should the libraries partition use for ApiCompat validation and similar.

**Review by MichalStrehovsky — COMMENTED** [review]
Would it make sense to also update the CI to use the new macro?

File: docs/workflow/building/coreclr/nativeaot.md [hunk 3]
```diff
@@ -2,6 +2,21 @@
 
 The Native AOT toolchain can be currently built for Linux (x64/arm64), macOS (x64) and Windows (x64/arm64).
 
+## Building
+
+1. [Install pre-requisites](../../README.md#build-requirements)
+1. Run `build[.cmd|.sh] clr.nativeaot+libs -rc [Debug|Release]` from the repo root to build binaries for local development. This will build individual components, but not the NuGet packages and builds much faster.
```
Comment (filipnavara) [inline docs/workflow/building/coreclr/nativeaot.md:8]: Should this really be `clr.nativeaot` and not `clr.aot`?

File: docs/workflow/building/coreclr/nativeaot.md (hunk 1 above)
Comment (agocke) [reply docs/workflow/building/coreclr/nativeaot.md:10]: I added a new subsection header. I think it's better here, as this is specifically for referencing binaries, not packages.

File: docs/workflow/building/coreclr/nativeaot.md [hunk 4]
```diff
@@ -2,6 +2,23 @@
...
 The Native AOT toolchain can be currently built for Linux (x64/arm64), macOS (x64) and Windows (x64/arm64).
 
+## Building
+
+1. [Install pre-requisites](../../README.md#build-requirements)
+1. Run `build[.cmd|.sh] clr.aot+libs -rc [Debug|Release]` from the repo root to build binaries for local development. This will build individual components, but not the NuGet packages and builds much faster.
+
+### Using built binaries
+
+The paths to major components can be overriden using `IlcToolsPath`, `IlcSdkPath`, `IlcFrameworkPath`, `IlcFrameworkNativePath` and `IlcMibcPath` properties for `dotnet publish`. For example, `/p:IlcToolsPath=<repo root>\artifacts\bin\coreclr\windows.x64.Debug\ilc` can be used to override the compiler with a local debug build for troubleshooting or quick iterations.
```
Comment (tlakollo) [inline docs/workflow/building/coreclr/nativeaot.md:12]:
```suggestion
The paths to major components can be overridden using `IlcToolsPath`, `IlcSdkPath`, `IlcFrameworkPath`, `IlcFrameworkNativePath` and `IlcMibcPath` properties for `dotnet publish`. For example, `/p:IlcToolsPath=<repo root>\artifacts\bin\coreclr\windows.x64.Debug\ilc` can be used to override the compiler with a local debug build for troubleshooting or quick iterations.
```

File: docs/workflow/building/coreclr/nativeaot.md [hunk 5]
```diff
@@ -2,6 +2,23 @@
...
+
+The component that writes out object files (objwriter.dll/libobjwriter.so/libobjwriter.dylib) is based on LLVM and doesn't build in the runtime repo. It gets published as a NuGet package out of the [dotnet/llvm-project](https://github.com/dotnet/llvm-project) repo (branch [objwriter/12.x](https://github.com/dotnet/llvm-project/tree/objwriter/12.x)). If you're working on ObjWriter or bringing up a new platform that doesn't have ObjWriter packages yet, as additional pre-requiresites you need to build objwriter out of that repo and replace the file in the output.
```
Comment (tlakollo) [inline docs/workflow/building/coreclr/nativeaot.md:14]:
```suggestion
The component that writes out object files (objwriter.dll/libobjwriter.so/libobjwriter.dylib) is based on LLVM and doesn't build in the runtime repo. It gets published as a NuGet package out of the [dotnet/llvm-project](https://github.com/dotnet/llvm-project) repo (branch [objwriter/12.x](https://github.com/dotnet/llvm-project/tree/objwriter/12.x)). If you're working on ObjWriter or bringing up a new platform that doesn't have ObjWriter packages yet, as additional pre-requisites you need to build objwriter out of that repo and replace the file in the output.
```

**Review by tlakollo — APPROVED** [review]
Looks good to me

**Pull request merged**